html/pages/syslog.inc.php

```diff
@@ -49,7 +49,12 @@
                         <select name="program" id="program" class="form-control input-sm">
                             <option value="">All Programs</option>
                                 <?php
-                                foreach (dbFetchRows('SELECT DISTINCT `program` FROM `syslog` ORDER BY `program`') as $data) {
+                                $sqlstatement = 'SELECT DISTINCT `program` FROM `syslog`';
+                                if (is_numeric($vars['device'])) {
+                                    $sqlstatement = $sqlstatement . ' WHERE device_id=' . $vars['device'];
```

Comment:
You need to pass variables to be used as prepared statements.

Like: https://github.com/EnzoZafra/librenms/blob/69f4e92142efe96f09cfc25722a26aac45e9ac95/html/includes/table/ports.inc.php#L140

You basically pass the variable into an array and pass that to dbFetchRows()